update dcode structure (#4713)
langchain-ai/docs: src/oss/deepagents/code/quickstart.mdx

---
title: Quickstart
sidebarTitle: Quickstart
description: Install Deep Agents Code, run your first task, and use interactive or non-interactive modes
---

Deep Agents Code (`dcode`) is a terminal coding agent built on the [Deep Agents SDK](/oss/deepagents/quickstart). This guide covers installation, your first task, daily interactive use, automation with piping, and LangSmith tracing. For a feature overview, see [Deep Agents Code overview](/oss/deepagents/code/overview). For `config.toml` and provider settings, see [Configuration](/oss/deepagents/code/configuration).

## Install and run your first task

<Steps>

    <Step title="Install and launch" icon="terminal">
        ```bash
        curl -LsSf https://langch.in/dcode | bash
        ```
    </Step>

    <Step title="Add provider credentials" icon="key">
        Deep Agents Code works with any tool-calling LLM. OpenAI, Anthropic, and Google are available out of the box.

        Use the `/auth` command to connect with a provider. See [Providers](/oss/deepagents/code/providers) for the full list and credential details.

        <Note>

>             Web search uses [Tavily](https://tavily.com). Add a key from `/auth` or set `TAVILY_API_KEY`. See [Enable web search](/oss/deepagents/code/credentials#enable-web-search-with-tavily).

Reviewer: ```suggestion
            Web search uses [Tavily](https://tavily.com). Add a key with `/auth`. See [Enable web search](/oss/deepagents/code/credentials#enable-web-search-with-tavily).
```

auth screen in app mentions api key alternative

        | `Ctrl+Left` / `Ctrl+Right` | Move cursor one word left/right |

        <Note>
            **macOS `Cmd+Left` / `Cmd+Right` / `Cmd+Delete`**

            Terminal emulators intercept `Cmd`-modified keys before they reach the running application, so Deep Agents Code never receives them directly. Instead, the terminal translates them into the readline shortcuts above.

            - **Ghostty:** Works out of the box. `Cmd+Left`, `Cmd+Right`, and `Cmd+Delete` are translated to `Ctrl+A`, `Ctrl+E`, and `Ctrl+U` by default.
            - **iTerm2:** Not bound by default. Add the following under **Settings → Profiles → Keys → Key Mappings** as `Send Text with vim special chars`:
                - `Cmd+Left` → `\x01` (Ctrl+A)
                - `Cmd+Right` → `\x05` (Ctrl+E)
                - `Cmd+Delete` → `\x15` (Ctrl+U)
            - **Terminal.app:** No native UI for this remap. Use the `Ctrl`-based shortcuts directly.

            Word-wise motion (`Option+Left` / `Option+Right`) is handled the same way: terminals send `Esc+b` / `Esc+f`, which Deep Agents Code interprets as word-left/right.
        </Note>
    </Accordion>
</AccordionGroup>

### External editor

> Press `Ctrl+X` or type `/editor` to compose prompts in an external editor. Deep Agents Code checks `$VISUAL`, then `$EDITOR`, then falls back to `vi` (macOS/Linux) or `notepad` (Windows). GUI editors (VS Code, Cursor, Zed, Sublime Text, Windsurf) automatically receive a `--wait` flag so Deep Agents Code blocks until you close the file.

Reviewer: > then falls back to `vi`

is this true? i thought it would be `nano`

Reviewer 2: Falls back to vi in my experience. i would selfishly love a nano fallback

> Press `Ctrl+X` or type `/editor` to compose prompts in an external editor. Deep Agents Code checks `$VISUAL`, then `$EDITOR`, then falls back to `vi` (macOS/Linux) or `notepad` (Windows). GUI editors (VS Code, Cursor, Zed, Sublime Text, Windsurf) automatically receive a `--wait` flag so Deep Agents Code blocks until you close the file.

Reviewer: ```suggestion
Press `Ctrl+X` or type `/editor` to compose prompts in an external editor. Deep Agents Code checks `$VISUAL`, then `$EDITOR`, then falls back to `vi` (macOS/Linux) or `notepad` (Windows). GUI editors (VS Code, Cursor, Zed, etc.) automatically receive a `--wait` flag so Deep Agents Code blocks until you close the file.
```

do people still use windsurf?

        # Use the curated safe-command list
        dcode -n "Build the project" -S recommended

        # Allow any shell command
        dcode -n "Fix the build" -S all
        ```
    </Accordion>
</AccordionGroup>

<Warning>
    **Use with caution.**

    `-S all` (or `--shell-allow-list all`) lets the agent execute arbitrary shell commands with no human confirmation.
</Warning>

## Trace with LangSmith

Enable [LangSmith](https://smith.langchain.com) tracing to see agent operations, tool calls, and decisions in a LangSmith project.

Run `/auth` and add your LangSmith API key. Tracing is enabled on the next launch and persists across sessions. See [Provider credentials](/oss/deepagents/code/credentials#use-%2Fauth-recommended) for details on the credential manager.

To customize the project name or configure tracing without the TUI, add keys to `~/.deepagents/.env` so tracing is enabled in every session without per-shell exports:

```bash title="~/.deepagents/.env"
LANGSMITH_TRACING=true
LANGSMITH_API_KEY=lsv2_...
LANGSMITH_PROJECT=optional-project-name  # Specify a project name or default to "deepagents-code"

> ```

Reviewer: nb/scope creep: i think this should be under an accordion

---
title: Quickstart
sidebarTitle: Quickstart
description: Install Deep Agents Code, run your first task, and use interactive or non-interactive modes
---

> Deep Agents Code (`dcode`) is a terminal coding agent built on the [Deep Agents SDK](/oss/deepagents/quickstart). This guide covers installation, your first task, daily interactive use, automation with piping, and LangSmith tracing. For a feature overview, see [Deep Agents Code overview](/oss/deepagents/code/overview). For `config.toml` and provider settings, see [Configuration](/oss/deepagents/code/configuration).

Reviewer: we're duplicating intros with overview page

i don't know that these pages should be split

Reviewer 2: With reposition of the Overview page, I think this is an appropriate split

Author: Let's keep them split - people may not expect the install/quickstart under overview.

Author: Only the first sentence is duplicated and I think that's probably ok

## Install and run your first task

<Steps>

    <Step title="Install and launch" icon="terminal">
        ```bash
        curl -LsSf https://langch.in/dcode | bash
        ```
    </Step>

    <Step title="Add provider credentials" icon="key">
        Deep Agents Code works with any tool-calling LLM. OpenAI, Anthropic, and Google are available out of the box.

        Use the `/auth` command to connect with a provider. See [Providers](/oss/deepagents/code/providers) for the full list and credential details.

        <Note>
            Web search uses [Tavily](https://tavily.com). Add a key from `/auth` or set `TAVILY_API_KEY`. See [Enable web search](/oss/deepagents/code/credentials#enable-web-search-with-tavily).
        </Note>
    </Step>

    <Step title="Choose a model (optional)" icon="cpu">
        Run `/model` inside a session to open the interactive switcher, or launch with `--model`:

        ```bash
        dcode --model anthropic:claude-opus-4-8

>         dcode --model openai:gpt-5.5

Reviewer 2: Do we really think this is worth including in the quickstart? @mdrxy 

I've never used dcode this way - and I wouldn't expect users to knowing that we have a nice TUI for it
I could see this being used - but maybe less relevant for the quickstart? happy to be disagreed with

Reviewer: > maybe less relevant for the quickstart

agree

        dcode --model fireworks:accounts/fireworks/models/deepseek-v4-pro
        dcode --model baseten:moonshotai/Kimi-K2.7-Code
        ```

        See [Model providers](/oss/deepagents/code/providers) for the full provider list, open weights options, and credential details.
    </Step>

    <Step title="Give the agent a task" icon="message">
        ```txt
        Create a Python script that prints "Hello, World!"
        ```

        The agent interprets the query and proposes changes with diffs for your approval before modifying files. If needed, it can run shell commands to test the code, check documentation, or search the web for up-to-date information.
    </Step>

    <Step title="Enable tracing (optional)" icon="chart-dots">

>         To log agent operations, tool calls, and decisions in LangSmith, add the following to `~/.deepagents/.env` or export the variables in your shell:

Reviewer: ```suggestion
    <Step title="Enable tracing" icon="chart-dots">
        To log agent operations, tool calls, and decisions in LangSmith, optionally add the following to `~/.deepagents/.env` or export the variables in your shell:
```

        ```bash title="~/.deepagents/.env"
        LANGSMITH_TRACING=true

>         LANGSMITH_API_KEY=lsv2_...

Reviewer 2: Similarly - we have a more delightful way to do this now via the TUI. better to push the delightful way? @mdrxy @npentrel
